File: 0547_Friend Circles/FriendCircles.cs
public class Solution {
    public int FindCircleNum(int[][] M) {
        if(M == null) return 0;
        var circles = 0;
        var visited = new bool[M.Length];
        
        for(int i=0;i<M.Length;i++){
            if(visited[i] == false){
                Check(M,i,visited);
                circles++;
            }
        }
        
        return circles;
    }
    
    private void Check(int[][] M, int current, bool[] visited)
    {
        for(int i=0;i<M[current].Length;i++){
            if(M[current][i] == 1 && visited[i] == false){
                visited[i] = true;
                Check(M,i,visited);
            }
        }
    }
}

File: 0547_Friend Circles/FriendCircles.java

public class FriendCircles {
    public int findCircleNum(int[][] M) {
        int circles = 0;
        boolean[] visited = new boolean[M.length];
        
        for(int i=0;i<M.length;i++){
        	if(visited[i] == false){
        		DFS(M, visited, i);
        		circles++;
        	}
        }
        
        return circles;
    }
    
    private void DFS(int[][] M, boolean[] visited, int current){
    	for(int j=0;j<M[current].length;j++){
    		if(M[current][j] == 1 && visited[j] == false){
    			visited[j] = true;
    			DFS(M, visited, j);
    		}
    	}
    }
}
